src/react/components/toolbar/toolbarItem.tsx

```diff
@@ -0,0 +1,56 @@
+import React from "react";
+import { IProject } from "../../../models/applicationState";
+import IProjectActions from "../../../redux/actions/projectActions";
+
+export interface IToolbarItemMetadata {
+    name: string;
+    icon: string;
+    tooltip: string;
+    group: string;
+    type: ToolbarItemType;
+}
+
+export enum ToolbarItemType {
+    Action = 0,
+    State = 1,
+}
+
+export interface IToolbarItemProps extends IToolbarItemMetadata, React.Props<ToolbarItem> {
+    key: string;
+    actions: IProjectActions;
+    project: IProject;
+    active: boolean;
+    onClick: (item: ToolbarItem) => void;
+}
+
+export abstract class ToolbarItem extends React.Component<IToolbarItemProps> {
+    constructor(props, context) {
+        super(props, context);
+
+        this.onItemClick = this.onItemClick.bind(this);
+        this.onClick = this.onClick.bind(this);
+    }
+
+    public render() {
+        const className = ["toolbar-btn"];
+        if (this.props.active) {
+            className.push("active");
+        }
+
+        return (
+            <button type="button"
+                className={className.join(" ")}
```

Comment:
Are we intentionally want to apply two css class here for the active button ?  I'm newbie here so this is just to understand but it shoudn;t be better to have a specific toolbar-btn-active class ?

Reply:
Yes, this in intentional, CSS is all about layering (cascading) multiple classes to achieve the desired results, otherwise you would end up with many duplicated styles across several different different CSS selectors.